### PR TITLE
fix: Ensure Response is mutable in Pages functions

### PR DESCRIPTION
**Pull request by sidharthachatterjee**
Fixes https://github.com/cloudflare/wrangler2/issues/1244

### CODE REVIEW

File: packages/wrangler/templates/pages-template-worker.ts [hunk 1]
```diff
@@ -126,19 +126,20 @@ export default {
           waitUntil: workerContext.waitUntil.bind(workerContext),
         };
 
-        const response = await handler(context);
+        const response: Response = await handler(context);
```
Comment (jrf0110) [inline packages/wrangler/templates/pages-template-worker.ts:129]: What if `handler` doesn't return an instance of `Response`? This is effectively just coercing an `any` to `Response`, which we should be pretty careful about. Maybe we can remove the type annotation and then perform an `instanceof` check instead?

```tsx
const response = await handler(context);

if (!(response instanceof Response)) {
  throw new Error('...')
}
```

Comment (sidharthachatterjee) [reply packages/wrangler/templates/pages-template-worker.ts:129]: Good thinking. I think I'm fairly certain we require handlers to return an instance of `Response` though?

Comment (petebacondarwin) [reply packages/wrangler/templates/pages-template-worker.ts:129]: AFAICT the `handler` is already of type: `PagesFunction<unknown, string, Record<string, unknown>>`, which has the following call-member:

```ts
(context: EventContext<Env, P, Data>) => Response | Promise<Response>
```

So calling `await handler()` should already be a `Response` not `any`. So there should be no need for the `: Response` type here and no need to do any casting.

File: packages/wrangler/templates/pages-template-worker.ts [hunk 2]
```diff
@@ -126,19 +126,20 @@ export default {
...
 
         // https://fetch.spec.whatwg.org/#null-body-status
-        return new Response(
-          [101, 204, 205, 304].includes(response.status) ? null : response.body,
-          { ...response, headers: new Headers(response.headers) }
-        );
+        return unlockResponse(response);
       } else if (__FALLBACK_SERVICE__) {
+        const response: Response = await env[__FALLBACK_SERVICE__].fetch(
```
Comment (petebacondarwin) [inline packages/wrangler/templates/pages-template-worker.ts:134]: Similarly, the `fetch()` method here is already typed to return a `Promise<Response>` so no need for the explicit type.

File: packages/wrangler/templates/pages-template-worker.ts [hunk 3]
```diff
@@ -149,3 +150,9 @@ export default {
     }
   },
 };
+
+const unlockResponse = (response: Response) =>
```
Comment (petebacondarwin) [inline packages/wrangler/templates/pages-template-worker.ts:154]: BIKESHED - unlock seems like a strange description for this function.

Comment (sidharthachatterjee) [reply packages/wrangler/templates/pages-template-worker.ts]: I accept your bike shed, haha. Renamed to `cloneResponse` which is what it actually does. The mutability is a desired side effect of this.

**Pull request merged**